### DISCUSSION TITLE
Modern Setting Design

### DISCUSSION THREAD

**Discussion by onesounds** [Flow-Launcher/Flow.Launcher#712] [Ideas]
![flowlauncher_setting_design01](https://user-images.githubusercontent.com/6903107/135125178-948719ad-2222-42bc-8859-777b21558cc6.jpg)
- Add Language tap and move.

![flowlauncher_setting_design02](https://user-images.githubusercontent.com/6903107/135125188-cac216a0-926c-4b2f-89ac-943fa166eeba.jpg)
- Plugin listing. I think it's not possible my skill. 

![flowlauncher_setting_design03](https://user-images.githubusercontent.com/6903107/135125191-78554ee8-041b-4bdf-959a-fa83633889eb.jpg)
- Theme Listing. is it possible? I think it is too heavy. (read all xaml in theme folder, and drawing mini color palette)
- if add more theme, expand mini palette area. 
I think people who like wox will don't like this. (too big, heavy, etc...)
If someone creates a rough structure, details can be attached.

**Comment by onesounds** [comment]
![image](https://user-images.githubusercontent.com/6903107/135310693-6b421ff4-2c97-41f2-9206-6c0fa8797bf4.png)
Work in progress. 

I can't work on the images one by one, so I'm using fluent icon(https://docs.microsoft.com/en-us/windows/apps/design/style/segoe-fluent-icons-font) font (399kb). If you have any plan of applying the design, I'd like to know it's okay.

**Comment by jjw24** [comment]
> Add Language tap and move.

sorry what do you mean tap and move? does it move to the language section when you tap on it ? 

> Plugin listing. I think it's not possible my skill.

This should be pointing to https://flow-launcher.github.io/docs/#/plugins, this will be just a url update in the code i can take care of it no problem.

**Reply by onesounds** [reply]
I guess I was tired when I wrote that. It means that language has been moved to the tap.
After working on it, I think the existing plug-in layout will be more convenient. I didn't want to put two tabs, but I think I should.

**Reply by taooceros** [reply]
> This should be pointing to https://flow-launcher.github.io/docs/#/plugins, this will be just a url update in the code i can take care of it no problem.

I think he means listing existing Plugins? The issue maybe the customized setting panel from Plugins.

I guess the easiest way is to create a container holding the user control created by Plugins with a way to collapse it. That will align with the current design.

**Comment by jjw24** [comment]
>Theme Listing. is it possible? I think it is too heavy. (read all xaml in theme folder, and drawing mini color palette)
if add more theme, expand mini palette area.
I think people who like wox will don't like this. (too big, heavy, etc...)
If someone creates a rough structure, details can be attached.

True, maybe just provide light customisations. I have  been thinking in terms showing how to theme, a dedicated page in the docs site will be easier to maintain and extend than making all the customisation options available via flow's settings. This way people who do not know C# can still contribute on the theming guide and write up tips.

**Reply by onesounds** [reply]
It is heavy and looks less useful than the workload. I'm looking for a way to look better.

The theme should be provided as follows.
1. Basic theme that the user does not feel the need to fix.
2. Theme that can be fixed as much as possible.
3. Theme that is easy to edit and apply.

Guide documents are also good, but due to the current structure, it would be better to have a sample theme with a lot of annotations. (I worked on a lot of other software themes, but this is the best.) Users open actual theme files as editors rather than opening guide documents.

**Reply by jjw24** [reply]
Ok makes sense

**Comment by jjw24** [comment]
> I can't work on the images one by one, so I'm using fluent icon(https://docs.microsoft.com/en-us/windows/apps/design/style/segoe-fluent-icons-font) font (399kb). If you have any plan of applying the design, I'd like to know it's okay.

I think this design is good and using fluent icon from ms is fine. I am keen to get this design added, can you create a seperate PR when you are ready please :)

any additional thoughts @Flow-Launcher/team ?

**Comment by Garulf** [comment]
Wow, Looks great! Is this a mock-up?

**Reply by onesounds** [reply]
Yeah. The first three are pictures.

**Comment by Zeroto521** [comment]
It's total win11 style.

![image](https://user-images.githubusercontent.com/25895405/135634811-ccfdb0f8-6e3b-46ab-94e2-045b35a09a14.png)

The title bar is missed. It's still win7 style.
I don't know whether it could transform to win11 style or not.

![image](https://user-images.githubusercontent.com/25895405/135635033-a27eaa08-9850-4413-a0b2-8ba390cdcf7b.png)

**Reply by onesounds** [reply]
I'm not sure if it's possible. I think I can make it pretty even if I don't do it, but I'm also trying to match the win11 style if possible.

**Reply by Zeroto521** [reply]
https://github.com/ModernFlyouts-Community/ModernFlyouts could be a good reference.

![image](https://user-images.githubusercontent.com/25895405/135700021-7865ba8b-abde-4a9e-9d2c-5ba64019daf6.png)

But I'm not sure the UI frame of flow-launcher is the same as ModernFlyouts.

**Comment by onesounds** [comment]
**Recommend Hotkey List**
- need Recommend Hot key list.(selectable). I think many people don't know how to hotkey setting.
- Reply please what you use as a hot key. personally I use WIN+Caps. I think most is  Twice ctrl, alt+space. 
- What is flow launcher default hotkey? I don't remember.

**Launcher window Fade In + Slide up animation.**
- When I tried, it has window blinking issue.
- It need because modern and it need where is launcher in screen. There were many cases where I had to find where.
- Configable. 
- Don't need fadout. same issue. just fast out.
- Other types of animations also need research.

**Pop up design and animation**
- It was very strange. It needs to be improve.

**ETC**
* Guide and Tips Tab.
* Plugin system is very important feature but pm install system is to hard most people (never used console)
There seems to be a way to improve further while using the same method.

**Reply by deefrawley** [reply]
> * need Recommend Hot key list.(selectable). I think many people don't know how to hotkey setting.

This has its own settings tab. Default is ALT + SPACE

> * Plugin system is very important feature but pm install system is to hard most people (never used console)

While it may be nice to be able to manage plugin install and update from the Settings window, using the command in the Flow search bar is kind of the whole point of Flow in the first place. If users can't get to grips with that then I'm not sure Flow is the tool for them and they probably accomplish the same things with other GUI tools.

**Reply by onesounds** [reply]
> using the command in the Flow search bar is kind of the whole point of Flow in the first place. 

I agree.

> If users can't get to grips with that then I'm not sure Flow is the tool for them and they probably accomplish the same things with other GUI tools.

I means, there are no command information inside program. To know this, I had to read how to use it outside the program. it is explained on git-hub / usage tips / find more plugins button. Even people familiar with the console cannot use it if they don't know the command.  I think I should explain it to you even if it's small in the setting window.

In addition, since plug-ins are important, I think it is necessary to see the entire list available, not a small list window. Now I feel like I'm looking around a large shopping mall only with a very small window.

**Reply by onesounds** [reply]
>  using the command in the Flow search bar is kind of the whole point of Flow in the first place.

I agree.

>  If users can't get to grips with that then I'm not sure Flow is the tool for them and they probably accomplish the same things with other GUI tools.

I means, there are no command information inside program. To know this, I had to read how to use it outside the program. it is explained on git-hub / usage tips / find more plugins button. (Is there a place I don't know?) Even people familiar with the console cannot use it if they don't know the command. I think I should explain it to you even if it's small in the setting window.

In addition, since plug-ins are important, I think it is necessary to see the entire list available, not a small list window. Now I feel like I'm looking around a large shopping mall only with a very small window.

**Reply by deefrawley** [reply]
> I means, there are no command information inside program. To know this, I had to read how to use it outside the program. it is explained on git-hub / usage tips / find more plugins button. (Is there a place I don't know?) Even people familiar with the console cannot use it if they don't know the command. I think I should explain it to you even if it's small in the setting window.
> 
> In addition, since plug-ins are important, I think it is necessary to see the entire list available, not a small list window. Now I feel like I'm looking around a large shopping mall only with a very small window.

Yep I understand now. Perhaps Flow could have a tooltip system where if you activate it but don't type anything, a list of tooltips is cycled as text in the search window, at a lower opacity? As soon as the user types something this clears. Something like

![image](https://user-images.githubusercontent.com/18594055/135703605-98eb25c1-e19d-438d-8a0b-48a41effbcef.png)

**Comment by onesounds** [comment]
I thought the same thing. 'Enter the file you want' or 'Search Type *.jpg'. but not good in appearance, and plug-ins are usually found in settings.

![image](https://user-images.githubusercontent.com/6903107/135703974-f8b8fec4-53a8-42da-a78c-fce37ffc0d72.png)
it's not final.(In this case, people may not know it have to put it in the query box.) Anyway, something like that seems necessary.

**Reply by deefrawley** [reply]
Even just making sure the keyword `help` leads a user directly to the webpage would be good. Currently it seems to default too far down the  results
![image](https://user-images.githubusercontent.com/18594055/135704152-941d95da-c812-456d-b2d3-6d83ee139df8.png)

**Reply by onesounds** [reply]
> Even just making sure the keyword `help` leads a user directly to the webpage would be good. Currently it seems to default too far down the results 

I agree that the ranking is low. I think that should just come out as a top priority. and There is the same problem with this. At first, I don't think I'll type "help" here. A game called event[0] comes to mind. At that time, I kept typing help. The game was a case of being alone in a spaceship with only a old console. 🤤

**Reply by jjw24** [reply]
yep having the tip up there definitely is good now that we have more real-estate to play with lol
